### ISSUE TITLE
Injecting `$VAR` in `job_start()` calls

### ISSUE THREAD

**Issue by LucHermitte** [vim/vim#1158]
I've tested that:

``` vim
:let job = job_start(['sh', '-c', 'echo "foo"'])
:echo job_info(job)
```

returns

```
{'status': 'dead', 'stoponexit': 'term', 'exitval': 0, 'exit_cb': 0, 'channel': channel 23 closed, 'process': 14090}
```

And that in the shell (UNIX)

``` sh
FOO="bar" sh -c 'echo $FOO'
```

works well.

However, 

``` vim
" neither
:let job = job_start(['FOO="bar" sh', '-c', 'echo "foo"'])
" nor
:let job = job_start(['FOO="bar"', 'sh', '-c', 'echo "foo"'])
" nor
:let job = job_start('(FOO="bar" && echo "foo")')
 " NB: I haven't even tried to display `$FOO`, yet.
```

give correct results. In all cases, I get an _exitval_ of 122.

While I can't be sure, it looks like a limitation of `job_start`. If there a workaround that works, it'd be interesting to document it.

**Comment by mattn** [comment]
Currently, no way to specify environment variables for the new process. However I think this is good idea.
And I prefer to specify with job-option like below.

``` vim
let job = job_start(['sh', '-c', 'echo $FOO'], {'envs': {'FOO':'bar'}})
```

I'll make patch for this in later.

**Comment by LucHermitte** [comment]
I also prefer the syntax you propose. -- injecting environment variables with the syntax I've tried earlier is not portable to Windows systems.

I had to use a temporary script generated on-the-fly to set the variables (synchronously and asynchronously). (Note, somehow my request is related to issue #1116 -- here I'm looking at alternative ways to _unset_ environment variables: I don't set them globally, but on-the-fly)
